providers/edge3/src/airflow/providers/edge3/executors/edge_executor.py

```diff
@@ -177,7 +177,30 @@ def _check_worker_liveness(self, session: Session) -> bool:
             .with_for_update(skip_locked=True)
             .filter(
                 EdgeWorkerModel.state.not_in(
-                    [EdgeWorkerState.UNKNOWN, EdgeWorkerState.OFFLINE, EdgeWorkerState.OFFLINE_MAINTENANCE]
+                    [
+                        EdgeWorkerState.UNKNOWN,
+                        EdgeWorkerState.OFFLINE,
+                        EdgeWorkerState.OFFLINE_MAINTENANCE,
+                        EdgeWorkerState.MAINTENANCE_MODE,
+                        EdgeWorkerState.MAINTENANCE_PENDING,
+                        EdgeWorkerState.MAINTENANCE_REQUEST,
+                    ]
+                ),
+                EdgeWorkerModel.last_update < (timezone.utcnow() - timedelta(seconds=heartbeat_interval * 5)),
+            )
+            .all()
+        )
+
+        lifeless_workers_maintenance: list[EdgeWorkerModel] = (
```

Comment:
Instead of making a second query and a second loop... can you use the original query and check the condition in the loop below?